### PR TITLE
[Spack] Update py-packages to the latest version

### PR DESCRIPTION
**Pull request by aandvalenzuela**

### CODE REVIEW

File: pip/requirements.txt [hunk 1]
```diff
@@ -281,7 +281,7 @@ tenacity==8.0.1
 #NO_AUTO_UPDATE:3: Force to use tensorflow 2.6.0; this should match the version in tensorflow-sources.spec
 tensorboard==2.6.0
 tensorflow==2.6.0
-tensorflow-estimator==2.6.0
+tensorflow-estimator==2.8.0
```
Comment (iarspider) [inline pip/requirements.txt]: I think we should keep this one at 2.6.0 as well, to match tensorflow. But could be wrong. @smuzaffar ?

Comment (aandvalenzuela) [reply pip/requirements.txt]: Let's try the test for 2.6.0 (tensorflow latest version is also 2.8.0)

Comment (smuzaffar) [reply pip/requirements.txt]: right, we need to keep it to 2.6.0 (to match the version in `tensorflow-sources`)

Comment (aandvalenzuela) [reply pip/requirements.txt]: Thank you!

**Comment by aandvalenzuela** [comment]
please test

File: pip/requirements.txt [hunk 2]
```diff
@@ -298,7 +298,7 @@ tqdm==4.62.2
 traitlets==5.1.0
 typed-ast==1.4.3
 typing-extensions==3.10.0.2
-typing==3.7.4.3
```
Comment (iarspider) [inline pip/requirements.txt:301]: Did we not drop `py3-typing` some time ago? If not, we should:

> NOTE: in Python 3.5 and later, the typing module lives in the stdlib, and installing this package has NO EFFECT

(from https://pypi.org/project/typing/)

Comment (smuzaffar) [reply pip/requirements.txt:301]: I do not remember such a change. Yes, we should drop it as it is picked up from python3 installations
```
[muzaffar@lxplus796 ~]$ source /cvmfs/cms.cern.ch/slc7_amd64_gcc10/external/py3-typing/3.7.4.3/etc/profile.d/init.sh
[muzaffar@lxplus796 ~]$ python3 -c 'import typing;print(typing.__file__)'
/cvmfs/cms.cern.ch/slc7_amd64_gcc10/external/python3/3.8.2-llifpc/lib/python3.8/typing.py
```

@aandvalenzuela , can you please remove it from requirements.txt and also remove all of its usages e.g. https://github.com/cms-sw/cmsdist/search?q=py3-typing

**Comment by aandvalenzuela** [comment]
please test

**Comment by smuzaffar** [comment]
@aandvalenzuela , I would suggest to try to build it locally first. These types of updates take too many iteration. In order to build locally you need to run following on cmsdev machines
```
git clone git@github.com:aandvalenzuela/cmsdist -b update-packages
git clone git@github.com:cms-sw/pkgtools -b V00-34-XX
./pkgtools/cmsBuild --weekly -a slc7_amd64_gcc10 -i work -j 10 build cmssw-tool-conf
```

you might get errors like the above `ERROR: Missing dependencies for py3-pylint (python3) found: py3-dill` . To fix this type of errors you need to add `Requires: py3-dill` in `cmsdist/pip/pylint.file` and rerun the `./pkgtools/cmsBuild .... `command again. If `dill` is missing in pip/requirements.txt then please add its entry with the proper version.

File: pip/requirements.txt [hunk 3]
```diff
@@ -124,14 +124,14 @@ joblib==1.0.1
...
+jupyter-console==6.4.3
 jupyter-core==4.7.1
 jupyter-packaging==0.10.4
 jupyterlab-pygments==0.1.2
 jupyter-server==1.10.2
 jupyter-server-mathjax==0.2.3
-jupyterlab-widgets==1.0.1
-keras==2.6.0
+jupyterlab-widgets==1.1.0
+keras==2.8.0
```
Comment (smuzaffar) [inline pip/requirements.txt:134]: @aandvalenzuela , looks like this should also match the version of tensorflow. So for now keep it at `2.6.0`

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
please test
looks lile pylint needs newer astroid

**Comment by smuzaffar** [comment]
please test for slc7_aarch64_gcc11

**Comment by smuzaffar** [comment]
please test for slc7_ppc64le_gcc11

**Review by smuzaffar — APPROVED** [review]
looks good to go in

**Comment by smuzaffar** [comment]
@aandvalenzuela , this looks good to go in, do you plan to update it ?

**Pull request merged**